src/main/webapp/queue.js

```diff
@@ -0,0 +1,182 @@
+// Initializes resources for reading/writing to Firestore
+const firebaseConfig = {
+  apiKey: config.apiKey, // eslint-disable-line no-undef
+  authDomain: 'lounge-95f01.firebaseapp.com',
+  databaseURL: 'https://lounge-95f01.firebaseio.com',
+  projectId: 'youtube-lounge',
+  storageBucket: 'youtube-lounge.appspot.com',
+  messagingSenderId: '681171972170',
+  appId: '1:681171972170:web:4c6526b8eb788af9d876b3',
+  measurementId: 'G-JSDHBSMHS3',
+};
+const app =
+    firebase.initializeApp(firebaseConfig); // eslint-disable-line no-undef
+db = firebase.firestore(app); // eslint-disable-line no-undef
+
+// Track realtime changes to the database and update the visual queue on change
+// hardcoded for one room for now, can access different rooms through
+// window.location.search property
+db.collection('rooms') // eslint-disable-line no-undef
+    .doc('47jGbulshBCjcc8YOt8a')
+    .collection('information')
+    .doc('queue').collection('videos')
```

Comment:
Is the accessing different rooms for users too or just testing? If it is just one room id per person, could you save the reference to the videos collection to keep from having to call through these links each time?

Reply:
So each room is assigned an ID (the '47jG...' string) which holds the queue of videos. That room is the storage space for all of the users within that room. When I write this to be more universal, I'll be parsing the window's URL to see what room ID was given to the page, validating that the room exists in Firestore, and then accessing the queue as needed. So, it is one room ID for all of the people within the room.